crates/plugins/mediator-coordination/src/forward/routing.rs

```diff
@@ -0,0 +1,121 @@
+use std::fs;
+
+use didcomm::{Message, PackEncryptedOptions, UnpackOptions};
+
+use crate::web::{error::MediationError, AppState};
+
+struct MessageStore {
+    dirpath: String,
+}
+impl MessageStore {
+    fn persist_msg(&self, content: String, inode: String) {
+        fs::create_dir_all(&self.dirpath).unwrap();
+        let file = format!("{}/{}.json",&self.dirpath, inode,);
+        fs::write(file, content).unwrap();
+    }
+}
```

Comment:
Are you persisting messages to raw files? This should be done to a MongoDB collection instead. Please inspect how things are persisted to the database in mediation coordination steps.